File: JavaScript_Courses/JavaScript_Basics/Preparation/04_For_Loop_Exercise/06_Oscars.js
function oscars(input) {
    let actorName = input[0];
    let actorPoints = Number(input[1]);
    let evaluatorsCount = Number(input[2]);

    for (let i = 0; i < evaluatorsCount * 2; i += 2) {
        let evaluatorName = input[i + 3];
        let evaluatorPoints = Number(input[i + 4]);

        actorPoints += evaluatorName.length * evaluatorPoints / 2;
        // console.log(actorPoints);
        if (actorPoints > 1250.50) {
            console.log(`Congratulations, ${actorName} got a nominee for leading role with ${actorPoints.toFixed(1)}!`);
            break;
        }
    }

    if (actorPoints <= 1250.50) {
        console.log(`Sorry, ${actorName} you need ${(1250.5 - actorPoints).toFixed(1)} more!`);
    }

}


// oscars(["Zahari Baharov",
//     "205",
//     "4",
//     "Johnny Depp",
//     "45",
//     "Will Smith",
//     "29",
//     "Jet Lee",
//     "10",
//     "Matthew Mcconaughey",
//     "39"]);

File: Python_Courses/Python_Basics/Python_PB_2023_04/04_For_Loop_Exercise/06_Oscars.py
POINTS_NEEDED = 1250.5

actor = input()
starting_points = float(input())
judges_count = int(input())

total_points = starting_points
for _ in range(judges_count):  # [1, 2, 3, 4]; # [0, 1, 2, 3]
    judge_name = input()
    judge_points = float(input())
    total_points += len(judge_name) * judge_points / 2
    if total_points > POINTS_NEEDED:
        print(f"Congratulations, {actor} got a nominee for leading role with {total_points :.1f}!")
        break
else:
    print(f"Sorry, {actor} you need {POINTS_NEEDED - total_points :.1f} more!")
